examples/sveltekit/svelte.config.js

```diff
@@ -1,7 +1,13 @@
 import adapter from '@sveltejs/adapter-auto';
+import preprocess from 'svelte-preprocess';
 
 /** @type {import('@sveltejs/kit').Config} */
 const config = {
+  preprocess: preprocess({
+    replace: [
+      ['import.meta.env.VERCEL_ANALYTICS_ID', JSON.stringify(process.env.VERCEL_ANALYTICS_ID)]
+    ]
+  }),
```

Comment:
Is the env var hard-coded? If there's a way to make it `VITE_VERCEL_ANALYTICS_ID` instead, Vite can take care of the replacement automatically. (In future we plan to have [better handling of env vars](https://github.com/sveltejs/kit/issues/4296) but that's the current state of play.)

Either way, it would be better to use [`@rollup/plugin-replace`](https://www.npmjs.com/package/@rollup/plugin-replace) here (it'll work as a Vite plugin) so that it applies to all modules, not just Svelte components

Reply:
Actually I think you could just use `define`, no need for `plugin-replace`: https://vitejs.dev/config/#define

Reply:
Trying to do this: 

```js
import adapter from '@sveltejs/adapter-auto';

/** @type {import('@sveltejs/kit').Config} */
const config = {
  kit: {
    adapter: adapter(),

    // Override http methods in the Todo forms
    methodOverride: {
      allowed: ['PATCH', 'DELETE']
    },
    vite: {
      define: {
        'import.meta.env.VERCEL_ANALYTICS_ID': process.env.VERCEL_ANALYTICS_ID
      }
    }
  }
};

export default config;
```

And it seems to be swapping the value correctly, but I get an error

```js


ReferenceError: REDACTED is not defined
--
21:19:44.164 | at file:///vercel/path0/.svelte-kit/output/server/entries/pages/__layout.svelte.js:59:21
21:19:44.165 | at Object.$render (file:///vercel/path0/.svelte-kit/output/server/chunks/index-5d6c164c.js:110:18)
21:19:44.165 | at file:///vercel/path0/.svelte-kit/output/server/index.js:62:97
21:19:44.165 | at $render (file:///vercel/path0/.svelte-kit/output/server/chunks/index-5d6c164c.js:110:18)
21:19:44.165 | at Object.render (file:///vercel/path0/.svelte-kit/output/server/chunks/index-5d6c164c.js:118:20)
21:19:44.165 | at render_response (file:///vercel/path0/.svelte-kit/output/server/index.js:844:29)
21:19:44.165 | at async respond$1 (file:///vercel/path0/.svelte-kit/output/server/index.js:1886:25)
21:19:44.166 | at async resolve (file:///vercel/path0/.svelte-kit/output/server/index.js:2148:105)
21:19:44.166 | at async Object.handle (file:///vercel/path0/.svelte-kit/output/server/chunks/hooks-3ad7760f.js:5:20)
```

Any thoughts?

Reply:
try this:

```diff
define: {
-  'import.meta.env.VERCEL_ANALYTICS_ID': process.env.VERCEL_ANALYTICS_ID
+  'import.meta.env.VERCEL_ANALYTICS_ID': JSON.stringify(process.env.VERCEL_ANALYTICS_ID)
}
```

(you could also remove the `import.meta.env.` prefix from both the `define` map and your code, if you're so inclined)

Reply:
That worked! https://github.com/vercel/vercel/pull/8002